CMSgov/mint-app
# Easi 3288/discussion tagging email

# EASI-3288
## Changes and Description

- Store a reference to the tagged user or possible solution in the HTMLMention struct ( so you don't have to query the database again)
- Email when a user is tagged in a discussion or in a reply
- Email to Solution POCs when a possible solution is tagged in a discussion or a in a reply
   - Added an email config so that emails to POCs can be turned on and off by environment (so that real pocs aren't emails in LLES)
   



## How to test this change
1.  Use the postman collection to create a model plan
2. Use the collection to make a discussion
3. Look in [mailcatcher](http://127.0.0.1:1085/) to verify that you received an email for every unique tagged user / solution
4. Use postman to make a reply.
5. Look in [mailcatcher](http://127.0.0.1:1085/) to verify that you received an email for every unique tagged user / solution 
6. Verify that the emails match the design specifications from [FIGMA](https://www.figma.com/file/8MfXzXzmKWv8vBjTasUIkD/MINT---Notifications?type=design&node-id=3079-3721&mode=design&t=yT9qNTGBUr4qdDgG-0)

If desired, you can also test this using the frontend to make tags, and verify that emails are sent as expected


Also, to look at just the emails, you can run `scripts/dev email` to see the new emails.


## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
